tests/platform/thermal_control_test_helper.py

```diff
@@ -0,0 +1,367 @@
+BASE_DIR = os.path.dirname(os.path.realpath(__file__))
+FILES_DIR = os.path.join(BASE_DIR, 'files')
+
+
+class BaseMocker:
+    """
+    @summary: Base class for thermal control data mocker
+
+    This base class defines the basic interface to be provided by base mocker. Mockers implemented by each
+    vendor must be a subclass of this base class.
+    """
+    # Mocker type dictionary. Vendor must register their concrete mocker class to this dictionary.
+    _mocker_type_dict = {}
+
+    def __init__(self, dut):
+        """
+        Constructor of a mocker.
+        :param dut: DUT object representing a SONiC switch under test.
+        """
+        self.dut = dut
+
+    def mock_data(self):
+        """
+        Generate mock data.
+        :return:
+        """
+        pass
+
+    def check_result(self, actual_data):
+        """
+        Check actual data with mocked data.
+        :param actual_data: A dictionary contains actual command line data. Key of the dictionary is the unique id
+                            of a line of command line data. For 'show platform fan', the key is FAN name. Value
+                            of the dictionary is a list of field values for a line.
+        :return: True if actual data match mocked data else False
+        """
+        pass
+
+    def deinit(self):
+        """
+        Destructor. Vendor specific clean up work should do here.
+        :return:
+        """
+        pass
+
+    @classmethod
+    def register_mocker_type(cls, name, mocker_type):
+        """
+        Register mocker type with its name.
+        :param name: Name of a mocker type. For example: FanStatusMocker.
+        :param mocker_type: Class of a mocker.
+        :return:
+        """
+        cls._mocker_type_dict[name] = mocker_type
+
+    @classmethod
+    def get_mocker_type(cls, name):
+        """
+        Get mocker type by its name.
+        :param name: Name of a mocker type. For example: FanStatusMocker.
+        :return: Class of a mocker.
+        """
+        return cls._mocker_type_dict[name] if name in cls._mocker_type_dict else None
+
+
+def mocker(type_name):
+    """
+    Decorator for register mocker type.
+    :param type_name: Name of a mocker type.
+    :return:
+    """
+    def wrapper(object_type):
+        BaseMocker.register_mocker_type(type_name, object_type)
+        return object_type
+    return wrapper
+
+
+@pytest.fixture
+def mocker_factory():
+    """
+    Fixture for thermal control data mocker factory.
+    :return: A function for creating thermal control related data mocker.
+    """
+    mockers = []
+
+    def _create_mocker(dut, mocker_name):
+        """
+        Create vendor specified mocker object by mocker name.
+        :param dut: DUT object representing a SONiC switch under test.
+        :param mocker_name: Name of a mocker type.
+        :return: Created mocker instance.
+        """
+        asic_type = dut.facts['asic_type']
```

Comment:
In general, this is not about asic_type. this is about platform.
e.g. two platforms with different peripheral hardware and drivers have the same asic model

Reply:
Yes. I think you mean that we should implement mock code based on different platform. For example, "x86_64-mlnx_msn3700-r0" and "x86_64-mlnx_msn2700-r0" should mock different thing. However, this require a big change in our test code. I am afraid we have not time to refactor the test code...

Reply:
think sn3700 and 2700 can share the same mocking code as long as this is the same platform provider.
In case Dell and Arista, as far as I remember they use same asic_type == 'broadcom'.
To be more clear, I suggest the following:

```
platform = dut.facts['platform']

if 'mlnx' in platform:
    # this is mellanox platform, so return mellanox mocker
```

Does this make sense?

Reply:
yes. Will do.

Reply:
fixed.